### PR TITLE
Add tests for registration

### PR DESCRIPTION
**Pull request by shayakrach**

### CODE REVIEW

**Review by AshkenYariv — APPROVED** [review]
Looks good Shayy!! Approved

**Review by edenmaor — APPROVED** [review]

**Review by assafg95 — APPROVED** [review]

File: conftest.py [hunk 1]
```diff
@@ -73,6 +116,63 @@ def user_2() -> User:
...
+    student_data = valid_user_data.copy()
+    student_data['role'] = Role.STUDENT
+    return student_data
+
+
+@pytest.fixture
+def valid_HR_data(valid_user_data):
+    HR_data = valid_user_data.copy()
+    HR_data['role'] = Role.HR
+    return HR_data
```
Comment (ormergi) [inline conftest.py]: Please change variables and function name to follow [python naming convention](https://www.python.org/dev/peps/pep-0008/#function-and-variable-names)

File: conftest.py [hunk 2]
```diff
@@ -73,6 +116,63 @@ def user_2() -> User:
...
+@pytest.fixture
+def invalid_passwords_user_data(valid_user_data, invalid_passwords):
+    invalid_user_data = create_invalid_objects_data(valid_objects_data=valid_user_data,
+                                                    invalid_field='password1',
+                                                    invalid_values=invalid_passwords)
+    return invalid_user_data
+
+
+@pytest.fixture
+def invalid_second_password_user_data(valid_user_data):
```
Comment (ormergi) [inline conftest.py]: I would rephrase it to `user_data_with_non_matching_passwords `
and other similar fixtures as well (from `invalid_..._user_data` to `user_data_with_...`)

File: conftest.py [hunk 3]
```diff
@@ -73,6 +116,63 @@ def user_2() -> User:
...
 
+@pytest.fixture
+def valid_user_data():
+    student_data = {'first_name': DEFAULT_FIRST_NAME,
+                    'last_name': DEFAULT_LAST_NAME,
+                    'email': DEFAULT_VALID_MAIL,
+                    'role': DEFAULT_USER_ROLE,
+                    'username': DEFAULT_VALID_USERNAME,
+                    'password1': DEFAULT_VALID_PASSWORD,
+                    'password2': DEFAULT_VALID_PASSWORD}
```
Comment (ormergi) [inline conftest.py]: Why wont you use `User` class instead?

Using a map with magic strings is a bad practice because If tomorrow we descried to change one of the properties name, you need to go over all its usages and manually change the string to the new name.
Also you need to remember (no body remember, ever) those magic strings in order to use it, also no auto complete by the IDE.

**Review by ormergi — CHANGES_REQUESTED** [review]
Hi,
PR description you can mention that it improve tests coverage for user registration feature.
About the tests, there is another user role which is the admin right? 
if so I would expect to test it as well.

File: conftest.py (hunk 1 above)
Comment (shayakrach) [reply conftest.py]: Do you mean changing HR to hr? @ormergi 
If so, we need to change it in other places as well. 
HR is an abbreviation and that is why it is capitalized.

File: conftest.py (hunk 3 above)
Comment (shayakrach) [reply conftest.py]: Because it appears in another team code (code that has been approved), I assumed it was a good practice @ormergi.

File: conftest.py (hunk 2 above)
Comment (shayakrach) [reply conftest.py]: Thanks, @ormergi. 
It sounds better.

File: conftest.py (hunk 3 above)
Comment (shayakrach) [reply conftest.py]: `User` does not have password1 and password2 fields for example.

Do you mean using the RegisterForm fields @ormergi ?

File: conftest.py (hunk 1 above)
Comment (ormergi) [reply conftest.py]: > Do you mean changing HR to hr?

yes

> i If so, we need to change it in other places as well. HR is an abbreviation and that is why it is capitalized.

For now change only what this PR adds

File: conftest.py (hunk 3 above)
Comment (ormergi) [reply conftest.py]: > Because it appears in another team code (code that has been approved), I assumed it was a good practice @ormergi.

Well, what other projects do is not always the best practice :slightly_smiling_face:

> `User` does not have password1 and password2 fields for example.
> Do you mean using the RegisterForm fields @ormergi ?

I meant to use the `User` class instead of a dictionary, but as we saw together the problem is that `RegisterForm`
expects a dict on its `data` argument instead of a class type, what makes it harder to test.
We can workaround this by adding method that converts `User` to dict, and use it when passing the user data to the register form.

Also, note that`RegisterForm` inherits from `UserCreationForm` which already provides `username`, `password1` and `password2` attributes, `RegisterForm` should have add the rest of the attribute the app requires on registration.
Due to the fact that we dont have time to change `RegisterForm` before the final demo, we will have to go with
the current solution you preset on this PR.

**Review by ormergi — APPROVED** [review]
Thanks!
